marketo/url.go

```diff
@@ -41,3 +42,10 @@ func constructURL(base string, path ...string) (*urlbuilder.URL, error) {
 
 	return link, nil
 }
+
+func updateURLWithID(url *urlbuilder.URL, id string) (*urlbuilder.URL, error) {
```

Comment:
Can you move the code which adds or removes `.json` to separate functions so that we can reuse it?